dom/src/main/scala/fs2/dom/MutationRecord.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.dom
+
+import org.scalajs.dom
+
+abstract class MutationRecord[F[_]] private[dom] {
+
+  def `type`: String
```

Comment:
I think in scala-js-dom we should change make this an enum (which is a compatible change) but it might become an incompatible change for us here. Too bad, we'll deal with it when it happens 🙃